src/types/mod.rs

```diff
@@ -2031,6 +2031,10 @@ pub enum DelegationEvent {
         /// Delegator's id
         delegator_id: DelegatorId,
     },
+    BakerRemoved {
+        /// Baker's id
+        baker_id: BakerId,
+    },
```

Comment:
Should we add `#[non_exhaustive]` to this enum and possibly others? As it is, adding variants is a breaking change (adding that attribute would also be a breaking change).

Reply:
On balance, I lean towards no. I would rather that code that pattern matches on `DelegationEvent` breaks and needs to be fixed on a version update than has a catch-all clause that was never intended to be used, but ends up being used accidentally.

Reply:
Sure thing